p2p/net/swarm/swarm_metrics.go

```diff
@@ -69,13 +69,34 @@ var (
 		},
 		[]string{"transport", "security", "muxer", "early_muxer", "ip_version"},
 	)
+	dialsPerPeer = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: metricNamespace,
+			Name:      "dials_per_peer",
+			Help:      "Number of addresses dialed per peer",
+			// to count histograms with integral values accurately the bucket needs to be
+			// very narrow around the integer value
+			Buckets: []float64{0, 0.99, 1, 1.99, 2, 2.99, 3, 3.99, 4, 4.99, 5},
```

Comment:
Not sure if a histogram is the right abstraction here. We can probably also improve the graph here:
<img width="840" alt="image" src="https://github.com/libp2p/go-libp2p/assets/1478487/5b118c6c-17d9-413e-9386-8e0ee4d5261a">

What about using a counter here (with label 1, 2, 3, 4, 5, more), and incrementing the respective counter directly?

We could then display this a pie chart, which would allows to easily see that X% of connections succeed on the first attempt, Y% on the second one, and so on. That would be more meaningful than percentiles, wouldn't it?